sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1082,9 +1083,348 @@ -(void) testPushNetworkTimeoutWithErroredOps
+    [todoTable insert:@{ @"id" : @"id2", @"text" : @"client" } completion:^(NSDictionary *item, NSError *error) {
+        [insertExpectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+
+    XCTAssertEqual([self performVanillaPush], 2, "Expected all inserted items to have been pushed to the server by now");
+
+    return synchronizedItemCount;
+}
+
+// Performs a push operation and returns the count of all client items pushed so far
+-(int) performVanillaPush
+{
+    // Define a filter that returns an appropriate server response for the requested item
+    MSTestFilter *filter = [MSTestFilter new];
+    MSClient *filteredClient = [client clientWithFilter:filter];
+
+    filter.ignoreNextFilter = YES;
+    filter.onInspectResponseData = ^(NSURLRequest *request, NSData *data) {
+        NSError *error = nil;
+        
+        NSDictionary *item = [filteredClient.serializer itemFromData:request.HTTPBody withOriginalItem:nil ensureDictionary:YES orError:&error];
+        XCTAssertNil(error);
+        
+        NSString *responseString = [NSString stringWithFormat:@"{\"id\": \"%@\", \"text\":\"server\"}", item[@"id"]];
+        NSData *pushResponse = [responseString dataUsingEncoding:NSUTF8StringEncoding];
+
+        return pushResponse;
+    };
+
+    // Perform push
+    XCTestExpectation *pushExpectation = [self expectationWithDescription:@"push expectation"];
+    [filteredClient.syncContext pushWithCompletion:^(NSError *error) {
+        XCTAssertNil(error, "Expected successful push");
+        [pushExpectation fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    return [self synchronizedItemCount];
+}
+
+// Returns the number of synchronized items, i.e. items that are present on the server as well as the client.
+// An item is synchronized if it was successfully pushed to the server from the client OR
+// if it was successfully pulled from the server to the client.
+-(int) synchronizedItemCount
+{
+    // Initialization
+    MSSyncTable *todoTable = [client syncTableWithName:TodoTableNoVersion];
+    MSQuery *query = [[MSQuery alloc] initWithSyncTable:todoTable];
+
+    // Number synchronized items
+    __block int numSynchronizedItems = 0;
+
+    XCTestExpectation *readExpectation = [self expectationWithDescription:@"read expectation"];
+
+    // Read from the table
+    [client.syncContext readWithQuery:query completion:^(MSQueryResult *result, NSError *error) {
+        
+        // Count the number of synchronized items
+        for (NSDictionary* item in result.items) {
+            if ([item[@"text"] isEqualToString:@"server"]) {
+                ++numSynchronizedItems;
+            }
+        }
+
+        [readExpectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    return numSynchronizedItems;
+}
+
+// Verify that the specified NSOperation has finished
+-(void) verifyFinished:(NSOperation *)operation
```

Comment:
we want to verify that it indeed finishes. waituntilfinish will block forever if the nsop hasn't finished.